Movecraft/src/main/java/net/countercraft/movecraft/async/rotation/RotationTask.java

```diff
@@ -195,150 +194,145 @@ protected void execute() {
             }
-            getCraft().getAudience().sendMessage(faceMessage);
-
-            craftsInWorld = CraftManager.getInstance().getCraftsInWorld(getCraft().getWorld());
-            for (Craft craft : craftsInWorld) {
-                if (!newHitBox.intersection(craft.getHitBox()).isEmpty() && craft != getCraft()) {
-                    //newHitBox.addAll(CollectionUtils.filter(craft.getHitBox(),newHitBox));
-                    //craft.setHitBox(newHitBox);
-                    if (Settings.Debug) {
-                        Bukkit.broadcastMessage(String.format("Size of %s hitbox: %d, Size of %s hitbox: %d", this.craft.getType().getStringProperty(CraftType.NAME), newHitBox.size(), craft.getType().getStringProperty(CraftType.NAME), craft.getHitBox().size()));
-                    }
-                    craft.setHitBox(craft.getHitBox().difference(oldHitBox).union(newHitBox));
-                    if (Settings.Debug){
-                        Bukkit.broadcastMessage(String.format("Hitbox of craft %s intersects hitbox of craft %s", this.craft.getType().getStringProperty(CraftType.NAME), craft.getType().getStringProperty(CraftType.NAME)));
-                        Bukkit.broadcastMessage(String.format("Size of %s hitbox: %d, Size of %s hitbox: %d", this.craft.getType().getStringProperty(CraftType.NAME), newHitBox.size(), craft.getType().getStringProperty(CraftType.NAME), craft.getHitBox().size()));
-                    }
-                    break;
-                }
+        } else {
+            if (farthestZ > 0) {
+                return I18nSupport.getInternationalisedComponent("Contact/Subcraft Rotate - South");
+            } else {
+                return I18nSupport.getInternationalisedComponent("Contact/Subcraft Rotate - North");
             }
         }
+    }
+
+    private void rotateEntitiesOnCraft(Location tOP) {
+        if (craft instanceof SinkingCraft
+                && craft.getType().getBoolProperty(CraftType.ONLY_MOVE_PLAYERS)
+                || !craft.getType().getBoolProperty(CraftType.MOVE_ENTITIES)) {
+            return;
+        }
+
+        Location midpoint = new Location(
+                craft.getWorld(),
+                (oldHitBox.getMaxX() + oldHitBox.getMinX())/2.0,
+                (oldHitBox.getMaxY() + oldHitBox.getMinY())/2.0,
+                (oldHitBox.getMaxZ() + oldHitBox.getMinZ())/2.0);
+
+        List<EntityType> entityList = List.of(EntityType.PLAYER, EntityType.PRIMED_TNT);
+        for(Entity entity : craft.getWorld().getNearbyEntities(midpoint,
+                oldHitBox.getXLength() / 2.0 + 1,
+                oldHitBox.getYLength() / 2.0 + 2,
+                oldHitBox.getZLength() / 2.0 + 1)) {
+
+            rotateHumanEntity(entity);
+
+            if (craft.getType().getBoolProperty(CraftType.ONLY_MOVE_PLAYERS)
+                    && (!entityList.contains(entity.getType())
+                    || craft instanceof SinkingCraft)) {
```

Comment:
Defining the following symbols:
- `a`: `craft.getType().getBoolProperty(CraftType.ONLY_MOVE_PLAYERS)`
- `b`: `entity.getType() == EntityType.PLAYER`
- `c`: `entity.getType() == EntityType.PRIMED_TNT`
- `d`: `craft instanceof SinkingCraft`
- `e`: `b || c` or `entityList.contains(entity.getType())`

Inverting the original `!a || ((b || c) && !d)` means this is correct as `a && (!e || d)`.

Reply:
yeah i fixed it in a commit it was before commit [3904a06](https://github.com/APDevTeam/Movecraft/pull/683/commits/3904a067c16aef8a7f3f9f45380613a3ced88b58)